scipy/stats/_continuous_distns.py

```diff
@@ -8572,6 +8572,111 @@ def _rvs_scalar(self, a, b, numsamples=None, random_state=None):
 truncnorm = truncnorm_gen(name='truncnorm', momtype=1)
 
 
+class truncpareto_gen(rv_continuous):
+    r"""An upper truncated Pareto continuous random variable.
+
+    %(before_notes)s
+
+    See Also
+    --------
+    pareto
+
+    Notes
+    -----
+    The probability density function for `truncpareto` is:
+
+    .. math::
+
+        f(x, b, c) = \frac{b}{1 - c^{-b}} \frac{1}{x^{b+1}}
+
+    for :math:`b > 0`, :math:`c > 1` and :math:`1 \le x \le c`.
+
+    `truncpareto` takes :math:`b`, :math:`c` as shape parameters.
+
+    Notice that the upper truncation value :math:`c` is defined in
+    standardized form:
+
+    .. math::
+
+        1 = (u_l - loc)/scale \quad ; \quad c = (u_r - loc)/scale
+
+    where :math:`u_l` and :math:`u_r` are the specific left and right
+    truncation values, respectively. In other words, the support of the
+    distribution becomes :math:`(scale + loc) <= x <= (c*scale + loc)` when
+    :math:`loc` and/or :math:`scale` are provided.
+
+    %(after_notes)s
+
+    References
+    ----------
+    .. [1] Burroughs, S. M., and Tebbens S. F.
+        "Upper-truncated power laws in natural systems."
+        Pure and Applied Geophysics 158.4 (2001): 741-757.
+
+    %(example)s
+
+    """
+
+    def _shape_info(self):
+        ib = _ShapeInfo("b", False, (0, np.inf), (False, False))
+        ic = _ShapeInfo("c", False, (0, np.inf), (False, False))
+        return [ib, ic]
+
+    def _argcheck(self, b, c):
+        return (b > 0) & (c > 1)
+
+    def _get_support(self, b, c):
+        return self.a, c
+
+    def _pdf(self, x, b, c):
+        return b * x**-(b+1) / (1 - c**-b)
+
+    def _logpdf(self, x, b, c):
+        return np.log(b) - np.log1p(-c**-b) - (b+1)*np.log(x)
+
+    def _cdf(self, x, b, c):
+        return (1 - x**-b) / (1 - c**-b)
+
+    def _logcdf(self, x, b, c):
+        return np.log1p(-x**-b) - np.log1p(-c**-b)
+
+    def _ppf(self, q, b, c):
+        return pow(1 - (1 - c**-b)*q, -1/b)
+
+    def _sf(self, x, b, c):
+        return (x**-b - c**-b) / (1 - c**-b)
+
+    def _logsf(self, x, b, c):
+        return np.log(x**-b - c**-b) - np.log1p(-c**-b)
+
+    def _isf(self, q, b, c):
+        return pow(c**-b + (1 - c**-b)*q, -1/b)
+
+    def _entropy(self, b, c):
+        return -(np.log(b/(1 - c**-b))
+                 + (b+1)*(np.log(c)/(c**b - 1) - 1/b))
+
+    def _munp(self, n, b, c):
+        if n == b:
+            return b*np.log(c) / (1 - c**-b)
+        else:
+            return b / (b-n) * (c**b - c**n) / (c**b - 1)
+
+    def _fitstart(self, data, args=None):
+        x0_ = pareto.fit(data)
```

Comment:
This is fine, and there's no need to check if you're not interested, but I'm curious whether the notes [here](https://github.com/scipy/scipy/issues/12545#issuecomment-735313963) can be extended to derive analytical fit including the truncation parameter?

Reply:
The Lagrange multipliers approach is pretty cool!
I had a quick look; it seems that the constraints on the domain reduce to (I use s for scale and tau for location)
![image](https://user-images.githubusercontent.com/34381942/171614681-a08e0902-0b4b-4d5d-9e4d-6612b46e17f0.png),
which gives
![image](https://user-images.githubusercontent.com/34381942/171614990-e5654d5d-980d-42d8-ad4b-1965e1d610fd.png).
We are left with an optimization problem on b and tau.

I will need to look into it in more details and double-check my derivations.

Reply:
> This is fine, and there's no need to check if you're not interested, but I'm curious whether the notes [here](https://github.com/scipy/scipy/issues/12545#issuecomment-735313963) can be extended to derive analytical fit including the truncation parameter.

There are definitely optimisations to be made regarding fitting. I have yet to check for combinations involving more than one fixed parameter, but for none or one provided, `root_scalar` should do the trick.

I'll address the requested changes and commit along the improved `fit` whenever I can.